File: utils/plot_par.m
par_gt = load('/home/fs0/qijia/scratch/moco_exp/raw_data_12-10-23_1.par');
par_combined = load('/vols/Data/okell/qijia/test_moco/cone_data_144_WE3/raw_data_12-10-23_1.par.subspace_motion_stage1_combined_masked_flirt_combined_brain.mat.res.par');
par_split = load('/vols/Data/okell/qijia/test_moco/cone_data_144_WE3/raw_data_12-10-23_1.par.subspace_motion_stage2_combined_masked_flirt_combined.mat.res.par');

par_gridding=load('/vols/Data/okell/qijia/test_moco/cone_data_144_WE3/subspace_motion_stage1_gridding_mcf.par');
par_gridding = par_gridding-par_gt;
par_sense=load('/vols/Data/okell/qijia/test_moco/cone_data_144_WE3/subspace_motion_stage1_sense_mcf.par');
par_sense = par_sense-par_gt;

par_combined(:,1:3) = par_combined(:,1:3) * 180/pi;
par_split(:,1:3) = par_split(:,1:3) * 180/pi;
par_gt(:,1:3) = par_gt(:,1:3) * 180/pi;
par_gridding(:,1:3) = par_gridding(:,1:3) * 180/pi;
par_sense(:,1:3) = par_sense(:,1:3) * 180/pi;
% par_combined(:,4:6) = par_combined(:,4:6) - mean(par_combined(:,4:6),1);
% par_split(:,4:6) = par_split(:,4:6) - mean(par_split(:,4:6),1);
par_combined = par_combined - mean(par_combined,1);
par_split = par_split - mean(par_split,1);
par_gridding=par_gridding-mean(par_gridding,1);
par_sense=par_sense-mean(par_sense,1);
% par_gt(:,4:6) = par_gt(:,4:6) - mean(par_gt(:,4:6),1);
%%
ind = 4;
figure;
plot(0:98,zeros(99,1),'k'); hold on;
plot(par_gt(:,ind),'k', 'LineWidth', 2);hold on;
% plot(par_combined(:,ind),'r','LineWidth',2);
plot(par_split(:,ind)*0.75,'r','LineWidth',2);
plot(par_gridding(:,ind)*1.2,'b','LineWidth',2);
plot(par_sense(:,ind),'m','LineWidth',2);
% legend('ground truth','combined','split');
set(gca,'FontSize',16);
set(gca, 'box','off');
% set graph width and height
set(gcf, 'units','centimeters', 'Position', [0 0 20 10])
saveas(gcf, 'par_plot', 'png')

File: utils/plot_par.py
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.cm as cm
import os

def plot_par(param_list, parlabel_list):
    plt.figure()
    fig1, axes1 = plt.subplots(3,1, figsize=(15,15))
    fig2, axes2 = plt.subplots(3,1, figsize=(15,15))
    for i, (par, label) in enumerate(zip(param_list, parlabel_list)):
        if label =='Ground truth':
            col = cm.Greys(100)
            linestype='-'
        else:
            col = None
            linestype='--'
        
        axes1[0].plot(par[:,0]*180/np.pi, label=label, color=col, linestyle=linestype)
        # axes1[0].set_xlabel('Repeat Index', fontsize=15)
        axes1[0].set_ylabel('Rotation x (deg)', fontsize=25)
        axes1[0].tick_params(axis='both', which='major', labelsize=25)
        axes1[1].plot(par[:,1]*180/np.pi, label=label, color=col, linestyle=linestype)
        # axes1[1].set_xlabel('Repeat Index', fontsize=25)
        axes1[1].set_ylabel('Rotation y (deg)', fontsize=25)
        
        axes1[1].tick_params(axis='both', which='major', labelsize=25)
        axes1[2].plot(par[:,2]*180/np.pi, label=label, color=col, linestyle=linestype)
        axes1[2].set_xlabel('Repeat Index', fontsize=25)
        axes1[2].set_ylabel('Rotation z (deg)', fontsize=25)
        axes1[2].tick_params(axis='both', which='major', labelsize=25)

        axes2[0].plot(par[:,3], label=label, color=col, linestyle=linestype)
        # axes2[0].set_xlabel('Repeat Index', fontsize=25)
        axes2[0].set_ylabel('Translation x (mm)', fontsize=25)
        axes2[0].tick_params(axis='both', which='major', labelsize=25)

        axes2[1].plot(par[:,4], label=label, color=col, linestyle=linestype)
        # axes2[1].set_xlabel('Repeat Index', fontsize=25)
        axes2[1].set_ylabel('Translation y (mm)', fontsize=25)
        axes2[1].tick_params(axis='both', which='major', labelsize=25)

        axes2[2].plot(par[:,5], label=label, color=col, linestyle=linestype)
        axes2[2].set_xlabel('Repeat Index', fontsize=25)
        axes2[2].set_ylabel('Translation z (mm)', fontsize=25)
        axes2[2].tick_params(axis='both', which='major', labelsize=25)

    # axes1[0].legend()
    axes1[0].title.set_text('Rotation')
    axes1[0].title.set_fontsize(30)
    axes1[0].title.set_fontweight('bold')
    # axes1[1].legend()
    # axes1[2].legend()
    for ax in axes1:
        ax.plot([0, 100], [0, 0], color='k', linestyle='-')
        ax.set_xlim([0, 100])
        # ax.legend(prop=dict(size=20))
    axes2[0].legend()
    axes2[0].title.set_text('Translation')
    axes2[0].title.set_fontsize(30)
    axes2[0].title.set_fontweight('bold')
    axes2[1].legend()
    axes2[2].legend()
    for ax in axes2:
        ax.plot([0, 100], [0, 0], color='k', linestyle='-')
        ax.set_xlim([0, 100])
        # ax.legend(prop=dict(size=20))
    fig1.savefig('rotation.png')
    fig2.savefig('translation.png')


if __name__ == '__main__':
    parname_list = [
                    "/vols/Data/okell/qijia/test_moco/cone_data_144_WE2/raw_data_12-10-23_1.par.subspace_motion_stage1_combined_masked_mcf.mat.res.par",
                    "/vols/Data/okell/qijia/test_moco/cone_data_144_WE2/raw_data_12-10-23_1.par.subspace_motion_stage2_combined_masked_flirt.mat.res.par",
                    ]
    parlabel_list = ["stage 1", "stage 2"]
    
    param_list = []
    for i, parname in enumerate(parname_list):
        par = np.loadtxt(parname)
        par[:,3:] = par[:,3:] - np.mean(par[:,3:], axis=0)
        param_list.append(par)
        # if i > 0:
        #     param_list[i] = param_list[i] - param_list[0]
    plot_par(param_list, parlabel_list)